crates/autopilot/src/domain/settlement/trade/mod.rs

```diff
@@ -0,0 +1,184 @@
+    pub fn uid(&self) -> &domain::OrderUid {
+        match self {
+            Self::Fulfillment(trade) => &trade.uid,
+            Self::Jit(trade) => &trade.uid,
+        }
+    }
+
+    /// Return JIT order if it's a JIT order.
+    pub fn as_jit(&self) -> Option<&Jit> {
+        match self {
+            Self::Fulfillment(_) => None,
+            Self::Jit(trade) => Some(trade),
+        }
+    }
+
+    /// CIP38 score defined as surplus + protocol fee
+    pub fn score(&self, auction: &super::Auction) -> Result<eth::Ether, math::Error> {
+        match self {
+            Self::Fulfillment(trade) => math::Trade::from(trade.clone()).score(auction),
+            Self::Jit(trade) => math::Trade::from(trade.clone()).score(auction),
+        }
+    }
+
+    /// Surplus of a trade.
+    pub fn surplus_in_ether(&self, prices: &auction::Prices) -> Result<eth::Ether, math::Error> {
+        match self {
+            Self::Fulfillment(trade) => math::Trade::from(trade.clone()).surplus_in_ether(prices),
+            Self::Jit(trade) => {
+                if trade.surplus_capturing {
+                    math::Trade::from(trade.clone()).surplus_in_ether(prices)
+                } else {
+                    // JIT orders that are not surplus capturing have zero
+                    // surplus, even if they settled at a better price than
+                    // limit price.
+                    Ok(eth::Ether::zero())
+                }
+            }
+        }
+    }
+
+    /// Total fee taken for the trade.
+    pub fn fee_in_ether(&self, prices: &auction::Prices) -> Result<eth::Ether, math::Error> {
+        match self {
+            Self::Fulfillment(trade) => math::Trade::from(trade.clone()).fee_in_ether(prices),
+            Self::Jit(trade) => math::Trade::from(trade.clone()).fee_in_ether(prices),
+        }
+    }
+
+    /// Total fee (protocol fee + network fee). Equal to a surplus difference
+    /// before and after applying the fees.
+    pub fn fee_in_sell_token(&self) -> Result<eth::SellTokenAmount, math::Error> {
+        match self {
+            Self::Fulfillment(trade) => math::Trade::from(trade.clone()).fee_in_sell_token(),
+            Self::Jit(trade) => math::Trade::from(trade.clone()).fee_in_sell_token(),
+        }
+    }
+
+    /// Protocol fees are defined by fee policies attached to the order.
+    pub fn protocol_fees_in_sell_token(
+        &self,
+        auction: &super::Auction,
+    ) -> Result<Vec<(eth::SellTokenAmount, fee::Policy)>, math::Error> {
+        match self {
+            Self::Fulfillment(trade) => {
+                math::Trade::from(trade.clone()).protocol_fees_in_sell_token(auction)
+            }
+            Self::Jit(trade) => {
+                math::Trade::from(trade.clone()).protocol_fees_in_sell_token(auction)
+            }
+        }
+    }
+
+    pub fn new(trade: transaction::EncodedTrade, auction: &super::Auction, created: u32) -> Self {
+        if auction.orders.contains_key(&trade.uid) {
```

Comment:
nanonit: small doc explaining why an order that is not in the auction is a jit order.